src/modules/bootloader/main.py

```diff
@@ -855,7 +884,12 @@ def prepare_bootloader(fw_type):
 
     efi_directory = libcalamares.globalstorage.value("efiSystemPartition")
 
-    if efi_boot_loader == "systemd-boot" and fw_type == "efi":
+    if efi_boot_loader == "clr-boot-manager":
+        if fw_type != "efi":
+            # Grub has to be installed first on non-EFI systems
+            install_grub(efi_directory, fw_type)
```

Comment:
We shouldn't need this since we already install `grub` when it isn't EFI at line 899.

Reply:
The `elif` chain won't fall through to that if `clr-boot-manager` is set as the bootloader in `bootloader.conf`, will it? Even if it does, CBM still has to be invoked after installing Grub. CBM is weird in that it will install systemd-boot for you, but it won't install Grub when it runs.

Clear Boot Manager manages both systemd-boot *and* Grub, it just for some reason only installs one of them.